From c6e45257db0b7bbb8ee73f59e74b72a4676fb73f Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 24 Oct 2014 20:55:30 -0400
Subject: [net] ipv4: Use IS_ERR_OR_NULL

Message-id: <92064c8d6db7f5233fe14bc2545f07bc5ef6d32a.1414183890.git.jbenc@redhat.com>
Patchwork-id: 98460
O-Subject: [RHEL7.1 net PATCH 1/5] net/ipv4: Use IS_ERR_OR_NULL
Bugzilla: 1144571
RH-Acked-by: Daniel Borkmann <dborkman@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Jesper Brouer <jbrouer@redhat.com>
RH-Acked-by: Hannes Frederic Sowa <hannes@redhat.com>
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1144571
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=8159555

commit 5a8dbf03ddf6dfe465b9f062e5b969f9606643f5
Author: Himangi Saraogi <himangi774@gmail.com>
Date:   Sun Jul 27 12:36:51 2014 +0530

    net/ipv4: Use IS_ERR_OR_NULL

    This patch introduces the use of the macro IS_ERR_OR_NULL in place of
    tests for NULL and IS_ERR.

    The following Coccinelle semantic patch was used for making the change:

    @@
    expression e;
    @@

    - e == NULL || IS_ERR(e)
    + IS_ERR_OR_NULL(e)
     || ...

    Signed-off-by: Himangi Saraogi <himangi774@gmail.com>
    Acked-by: Julia Lawall <julia.lawall@lip6.fr>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/ipv4/gre_offload.c b/net/ipv4/gre_offload.c
index a131b01..d3fe2ac 100644
--- a/net/ipv4/gre_offload.c
+++ b/net/ipv4/gre_offload.c
@@ -74,7 +74,7 @@ static struct sk_buff *gre_gso_segment(struct sk_buff *skb,
  /* segment inner packet. */
  enc_features = skb->dev->hw_enc_features & netif_skb_features(skb);
  segs = skb_mac_gso_segment(skb, enc_features);
- if (!segs || IS_ERR(segs)) {
+ if (IS_ERR_OR_NULL(segs)) {
   skb_gso_error_unwind(skb, protocol, ghl, mac_offset, mac_len);
   goto out;
  }
-- 
1.7.1